internal/resource/config/localidentity/local_identity_identity_profile_data_source.go

```diff
@@ -0,0 +1,671 @@
+package localidentity
+
+import (
+	"context"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/int64validator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	client "github.com/pingidentity/pingfederate-go-client/v1125/configurationapi"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/common/id"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/common/resourcelink"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/config"
+	internaltypes "github.com/pingidentity/terraform-provider-pingfederate/internal/types"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &localIdentityIdentityProfilesDataSource{}
+	_ datasource.DataSourceWithConfigure = &localIdentityIdentityProfilesDataSource{}
+)
+
+// Create a Administrative Account data source
+func NewLocalIdentityIdentityProfilesDataSource() datasource.DataSource {
+	return &localIdentityIdentityProfilesDataSource{}
+}
+
+// localIdentityIdentityProfilesDataSource is the datasource implementation.
+type localIdentityIdentityProfilesDataSource struct {
```

Comment:
nit: I'd rename these structs to be non-plural, since we are going to have a plural version of the data source eventually